src/main.rs

```diff
@@ -238,6 +238,17 @@ fn main() {
         std::process::exit(exitcode::CONFIG);
     });
 
+    // Start listening for config changes immediately.
+    // There is a chance of config file changing since we read
```

Comment:
do you mean since we _don't_ read it until the end of the fn?

That makes sense, I wonder if the easy fix is to read it first then call this function?

Reply:
I meant if change happens between 
```
    let config = vector::topology::Config::load(file);
```
and
```
vector::topology::config::watcher::config_watcher(...)
```
function calls. Then the change will probably not be detected.

Fixing it is easy as moving call to `fn config_watcher` before call to `Config::load`, but `config_watcher` needs `reload_config` flag to know if it should start listening and raising SIGHUPs, and we get that only with reading config file therefore after `Config::load`. The simplest solution is to accept `reload_config` flag as argument during Vector starting instead of it being a global option in `vector.toml`.  Then we could know to start watching for config changes before we start reading it. `-w` seems like a good choice for the flag.

Reply:
Yes, I think this makes sense, I would like to see this change :+1: